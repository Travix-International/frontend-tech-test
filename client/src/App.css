/* .App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */

#root {
  background-color:black;
}

.appContainer {
  display: flex;
  flex-direction: column;
  height: 100vh;
  overflow-y: hidden;
}

.alertContainer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.alertContainer .alert {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.addTaskContainer{
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 5vh
}

.sortFilterContainer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 5vh
}

.taskListContainer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 70vh
}