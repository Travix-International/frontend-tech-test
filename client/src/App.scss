@import './styles/mixin.scss';
@import './styles/default.scss';
@import './styles/theme.scss';

.app {
    height: 100%;
    background-color: $color-body-bg;
    padding: 0;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    @include screen(sm) {
        padding: 3rem;
    }
}


.bgText {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    
    p {
        margin: 0;
        font-size: 30vw;
        font-family: Arial, Helvetica, sans-serif;
        font-style: italic;
        color: $color-body-text;
    }
}