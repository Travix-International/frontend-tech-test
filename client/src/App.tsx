import React from 'react';
import styles from './App.scss';

class App extends React.Component<{}> {
    render() {
        return (
            <div className="App">
                <h1 className={styles.title}>Hello World!!</h1>
            </div>
        );
    }
}

export default App;