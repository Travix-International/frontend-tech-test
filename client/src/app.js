import * as React from 'react';
import { Provider } from 'react-redux';
import logger from 'redux-logger';
import PromiseMiddleware from 'redux-promise-middleware';
import { Grid, Row } from 'react-bootstrap';

import createStore from './store';
import reducers from './modules/reducers';
import ListTasks from './modules/tasks/list';


const store = createStore({ reducers, middlewares: [new PromiseMiddleware(), logger] });

export default class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Provider store={store}>
          <Grid>
            <Row>
              <ListTasks />
            </Row>
          </Grid>
        </Provider>
      </div>
    );
  }
}
