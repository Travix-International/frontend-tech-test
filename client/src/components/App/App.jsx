import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';

import TaskDetails from '../../containers/TaskDetails/TaskDetails';
import TaskManager from '../../containers/TaskManager/TaskManager';
import Container from '../UI/Container/Container';
import styles from './App.module.scss';
import routes from '../../constants/routes';


const App = () => (
  <div className={styles.App}>
    <Container>
      <h1>
        <Link className={styles.Logo} to="/">Task Manager</Link>
      </h1>
      <Switch>
        <Route component={TaskDetails} exact path={routes.taskPath} />
        <Route component={TaskManager} exact path={routes.tasks} />
      </Switch>
    </Container>
    
  </div>
);

export default App;
