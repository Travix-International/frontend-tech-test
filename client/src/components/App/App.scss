@import url('https://fonts.googleapis.com/css?family=Sawarabi+Mincho');
@import '../../scss/vars.scss';
body {
  background-color: $yellow;
  font-size: 16px;
  font-family: 'Sawarabi Mincho', sans-serif;
}