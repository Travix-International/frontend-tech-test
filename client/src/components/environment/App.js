import React, { Component } from 'react'
import { Provider } from 'react-redux'

import Tasks from '../ecosystems/Tasks/index'
import StoreConfigurator from '../../ducks/StoreConfigurator'

const storeConfigurator = new StoreConfigurator()
const store = storeConfigurator.configure()

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Tasks />
      </Provider>
    )
  }
}
