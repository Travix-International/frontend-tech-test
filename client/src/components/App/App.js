import React from 'react'
import LayoutContainer from '../../containers/LayoutContainer/LayoutContainer'

import './App.scss'

const App = () => <LayoutContainer />

export default App
