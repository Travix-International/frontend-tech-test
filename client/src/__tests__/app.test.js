import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../app';


describe('App', () => {
  it('should render', () => {
    const appElement = shallow(<App />);
    expect(toJson(appElement)).toMatchSnapshot();
  });
});
