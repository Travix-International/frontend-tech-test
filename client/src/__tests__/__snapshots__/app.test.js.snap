// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  className="App"
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Connect(ListTasks) />
      </Row>
    </Grid>
  </Provider>
</div>
`;
